js/PaymentRequest/PaymentRequestUpdateEvent.js

```diff
@@ -59,7 +59,7 @@ export default class PaymentRequestUpdateEvent {
 
     validateTotal(value.total, DOMException);
     validateDisplayItems(value.displayItems, DOMException);
-    validateShippingOptions(value.shippingOptions, DOMException);
+    validateShippingOptions(value, DOMException);
```

Comment:
Nice catch, not sure how this didn't expose problems before. Unfortunate that it's different than other `validate*`, but this one seems to want to modify your details in certain conditions.

Is it fine that we will be mutating `value`?

Reply:
For the scope of this PR, I think it's fine that the library mutates value. That's clearly how the original author intended it to work, and the fact that the wrong value was being passed to validateShippingOptions seems like simply a mistake. If the client needs to  preserve the exact state of the object before passing them in, the client can clone the input themselves and pass in the clone.

Reply:
For background, value comes from the client of the library calling updateWith on an event in one of the event listeners like `paymentmethodchanged` or `shippingaddresschanged`.

The original author intended to mutate the input, and the fact that they were passing the wrong object to validateShippingOptions seems like a simple mistake. Whether or not that is an ideal design decision is another matter, and is out of scope for this PR.

I think that it's fine for this library to essentially take ownership of objects once they are passed in and to do whatever it likes with them. If the client needs to preserve the exact state of the inputs, the client can clone the object before passing it in.